libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::abs;
+use crate::bignum::utils::recast_vec;
+use crate::bignum::Bignum;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        let mut is_neg: bool = false;
+        let lhs: Bignum;
+        let rhs: Bignum;
+        if !self.is_neg && other.is_neg {
+            return self + abs(&other);
+        } else if self.is_neg && !other.is_neg {
+            return self + (-other);
+        }
+        // make rhs the bigger number
+        if abs(&self) > abs(&other) {
+            if self.is_neg {
+                is_neg = true;
+            }
+            lhs = other;
+            rhs = self;
+        } else {
+            // this really is !self.is_neg && !other.is_neg but we can use demorgans for speed
+            if !(self.is_neg || other.is_neg) {
+                is_neg = true;
+            }
+            lhs = self;
+            rhs = other;
+        }
+
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let self_size: usize = lhs.dec.len();
+        let other_size: usize = rhs.dec.len();
+        let (res_int, lhs_int, res_dec, lhs_dec) = if self_size > other_size {
+            let mut equal_dec = vec![0; self_size];
+            equal_dec[..other_size].clone_from_slice(&rhs.dec[..other_size]);
+            (rhs.int, lhs.int, equal_dec, lhs.dec)
+        } else {
+            let mut equal_dec = vec![0; other_size];
+            equal_dec[..self_size].clone_from_slice(&lhs.dec[..self_size]);
+            (rhs.int, lhs.int, rhs.dec, equal_dec)
+        };
+
+        // now we need to make the bigger number the res dec
+        // 1. Decimal
+
+        let mut res: i8;
+        let mut carry: i8 = 0;
+        let mut res_dec_asi8 = recast_vec::<u8, i8>(res_dec);
+        let lhs_dec_asi8 = recast_vec::<u8, i8>(lhs_dec);
+        let mut res_int_asi8 = recast_vec::<u8, i8>(res_int);
+        let lhs_int_asi8 = recast_vec::<u8, i8>(lhs_int);
```

Comment:
I think it would make more sense to do this in lines 39-46.

Reply:
yep. will do